beaconclient/multi_beacon_client.go

```diff
@@ -180,3 +182,27 @@ func (c *MultiBeaconClient) beaconInstancesByLastResponse() []IBeaconInstance {
 
 	return instances
 }
+
+// PublishBlock publishes the signed beacon block via https://ethereum.github.io/beacon-APIs/#/ValidatorRequiredApi/publishBlock
+func (c *MultiBeaconClient) PublishBlock(block *types.SignedBeaconBlock) (code int, err error) {
+	log := c.log.WithFields(logrus.Fields{
+		"slot":      block.Message.Slot,
+		"blockHash": block.Message.Body.ExecutionPayload.BlockHash.String(),
+	})
+
+	clients := c.beaconInstancesByLastResponse()
+	for _, client := range clients {
+		log := log.WithField("uri", client.GetURI())
+		log.Debug("publishing block")
+
+		if code, err = client.PublishBlock(block); err != nil {
+			log.WithField("statusCode", code).WithError(err).Error("failed to publish block")
+			continue
+		}
+
+		log.WithField("statusCode", code).Info("published block")
+		return code, nil
+	}
+
+	return code, err
```

Comment:
not sure what you mean -- this would currently return the last error, by the last used BN 🤔 would indicate that the block wasn't published by any of them

Reply:
or do you mean it should return a specific, different error?